src/ORM/ResultSet.php

```diff
@@ -447,7 +448,7 @@ protected function _getTypes($table, $fields)
     {
         $types = [];
         $schema = $table->getSchema();
-        $map = array_keys(Type::map() + ['string' => 1, 'text' => 1, 'boolean' => 1]);
+        $map = array_keys((array)Type::map() + ['string' => 1, 'text' => 1, 'boolean' => 1]);
```

Comment:
Using Type::map() as getter is deprecated.
Use Type::getMap() instead.

Reply:
I wonder why deprecation warning isn't thrown when running tests.

Reply:
because you can not find  any usage of _getTypes in core :)
and this method removed in cake 4.0